middleware/kubernetes/kubernetes.go

```diff
@@ -384,9 +385,25 @@ func (k *Kubernetes) findPods(namespace, podname string) (pods []pod, err error)
 		return pods, nil
 	}
 
-	// TODO: implement cache verified pod responses
+	// PodModeVerified
+	podList, err := k.APIConn.podLister.List(labels.Everything())
```

Comment:
Probably? There is an indexing feature built in to the k8s api cache controller.  Not sure how it works yet. Will look into it.